client/src/utils/BGraphUtil.ts

```diff
@@ -54,24 +54,31 @@ export const loadBGraphData = (nodesPath: string, edgesPath: string): Promise<an
 };
 
 // Add filter terms here to change the order in which they are processed in bgraph
-const NODE_PRIORITY_RANK = 1;
-const EDGE_PRIORITY_RANK = 4;
+const EDGE_PRIORITY_RANK = 2;
+const NODE_PRIORITY_RANK = 4;
+const PATH_PRIORITY_RANK = 7;
 const filterTermToPriorityRank = {
+  // Edge Terms
+  [QUERY_FIELDS_MAP.BIO_EDGE_PRE.field]: 1,
+  [QUERY_FIELDS_MAP.BIO_EDGE_TESTED.field]: EDGE_PRIORITY_RANK,
+  [QUERY_FIELDS_MAP.BIO_EDGE_TYPE.field]: EDGE_PRIORITY_RANK,
+  [QUERY_FIELDS_MAP.BIO_EDGE_DOI.field]: EDGE_PRIORITY_RANK,
+  [QUERY_FIELDS_MAP.BIO_EDGE_POST.field]: 3,
   // Node Terms
-  [QUERY_FIELDS_MAP.BIO_NODE_PRE.field]: 0,
+  [QUERY_FIELDS_MAP.BIO_NODE_PRE.field]: 4,
   [QUERY_FIELDS_MAP.BIO_NODE_NAME.field]: NODE_PRIORITY_RANK,
   [QUERY_FIELDS_MAP.BIO_NODE_GROUP.field]: NODE_PRIORITY_RANK,
   [QUERY_FIELDS_MAP.BIO_NODE_GROUNDED.field]: NODE_PRIORITY_RANK,
   [QUERY_FIELDS_MAP.BIO_NODE_GROUNDED_ONTO.field]: NODE_PRIORITY_RANK,
   [QUERY_FIELDS_MAP.BIO_NODE_IN_DEGREE.field]: NODE_PRIORITY_RANK,
   [QUERY_FIELDS_MAP.BIO_NODE_OUT_DEGREE.field]: NODE_PRIORITY_RANK,
-  [QUERY_FIELDS_MAP.BIO_NODE_POST.field]: 2,
-  // Edge Terms
-  [QUERY_FIELDS_MAP.BIO_EDGE_PRE.field]: 3,
-  [QUERY_FIELDS_MAP.BIO_EDGE_TESTED.field]: EDGE_PRIORITY_RANK,
-  [QUERY_FIELDS_MAP.BIO_EDGE_TYPE.field]: EDGE_PRIORITY_RANK,
-  [QUERY_FIELDS_MAP.BIO_EDGE_DOI.field]: EDGE_PRIORITY_RANK,
-  [QUERY_FIELDS_MAP.BIO_EDGE_POST.field]: 5,
+  [QUERY_FIELDS_MAP.BIO_NODE_POST.field]: 6,
+  // Path Terms
+  [QUERY_FIELDS_MAP.BIO_PATH_PRE.field]: 0,
+  [QUERY_FIELDS_MAP.BIO_PATH_PRE_EDGE_FILTER_LOOP.field]: 1,
```

Comment:
just double checking, you sure you want `BIO_PATH_PRE_EDGE_FILTER_LOOP` and `BIO_PATH_PRE_NODE_FILTER_LOOP`  to have the same rank as `BIO_EDGE_PRE` and `BIO_EDGE_POST` respectively? order is not guaranteed when the ranks are the same

Reply:
I've been thinking of the architecture such that the PRE/POST fields of each query type `node`, `edge` and `path` are not meant to be used in multiple query type builders. So this shouldn't be an issue.

Reply:
just making sure this was intentional, if it is then its fine